File: backend/src/main/java/com/bika/company/service/DepartmentService.java
package com.bika.company.service;

import com.bika.company.dto.DepartmentDTO;
import com.bika.company.entity.Department;
import java.util.List;
import java.util.Optional;

public interface DepartmentService {
    DepartmentDTO createDepartment(DepartmentDTO departmentDTO);
    DepartmentDTO updateDepartment(Long id, DepartmentDTO departmentDTO);
    void deleteDepartment(Long id);
    DepartmentDTO getDepartmentById(Long id);
    List<DepartmentDTO> getAllDepartments();
    List<DepartmentDTO> getDepartmentsByCompany(Long companyId);
    Optional<Department> findEntityById(Long id);
} 

File: frontend/src/services/api/departmentService.ts
import apiClient from './client';
import { Department } from './types';

class DepartmentService {
  async getDepartments(): Promise<Department[]> {
    const response = await apiClient.get<Department[]>('/v1/departments');
    return response.data;
  }

  async getDepartmentsByCompany(companyId: number): Promise<Department[]> {
    const response = await apiClient.get<Department[]>(`/v1/departments/company/${companyId}`);
    return response.data;
  }

  async getDepartmentById(id: number): Promise<Department> {
    const response = await apiClient.get<Department>(`/v1/departments/${id}`);
    return response.data;
  }

  async createDepartment(departmentData: Partial<Department>): Promise<Department> {
    const response = await apiClient.post<Department>('/v1/departments', departmentData);
    return response.data;
  }

  async updateDepartment(id: number, departmentData: Partial<Department>): Promise<Department> {
    const response = await apiClient.put<Department>(`/v1/departments/${id}`, departmentData);
    return response.data;
  }

  async deleteDepartment(id: number): Promise<void> {
    await apiClient.delete(`/v1/departments/${id}`);
  }
}

const departmentService = new DepartmentService();
export default departmentService; 